sycl/test/check_device_code/fpga_ihs_float.cpp

```diff
@@ -395,55 +368,47 @@ void ap_float_pown() {
   // CHECK: call spir_func signext i15 @_Z{{[0-9]+}}__spirv_ArbitraryFloatPowNINTEL{{.*}}(i12 signext {{[%a-z0-9.]+}}, i32 7, i10 signext {{[%a-z0-9.]+}}, i1 zeroext false, i32 9, i32 0, i32 2, i32 1)
 }
 
-template <typename name, typename Func>
-__attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
-  kernelFunc();
-}
-
-int main() {
-  kernel_single_task<class kernel_function>([]() {
-    ap_float_cast<11, 28, 9, 30>();
-    ap_float_cast_from_int<43, 8, 16>();
-    ap_float_cast_to_int<7, 15, 30>();
-    ap_float_add<5, 7, 6, 8, 4, 9>();
-    ap_float_add<6, 8, 4, 9, 5, 7>();
-    ap_float_sub<4, 4, 5, 5, 6, 6>();
-    ap_float_mul<16, 34, 16, 34, 16, 34>();
-    ap_float_div<4, 11, 4, 11, 5, 12>();
-    ap_float_gt<20, 42, 21, 41>();
-    ap_float_ge<19, 27, 19, 27>();
-    ap_float_lt<2, 2, 3, 3>();
-    ap_float_le<27, 27, 26, 28>();
-    ap_float_eq<7, 12, 7, 7>();
-    ap_float_recip<9, 29, 9, 29>();
-    ap_float_rsqrt<12, 19, 13, 20>();
-    ap_float_cbrt<0, 1, 0, 1>();
-    ap_float_hypot<20, 20, 21, 21, 19, 22>();
-    ap_float_sqrt<7, 7, 8, 8>();
-    ap_float_log<30, 19, 19, 30>();
-    ap_float_log2<17, 20, 18, 19>();
-    ap_float_log10<4, 3, 4, 5>();
-    ap_float_log1p<17, 30, 18, 30>();
-    ap_float_exp<16, 25, 16, 25>();
-    ap_float_exp2<1, 1, 2, 2>();
-    ap_float_exp10<8, 16, 8, 16>();
-    ap_float_expm1<21, 42, 20, 41>();
-    ap_float_sin<14, 15, 16, 17>();
-    ap_float_cos<1, 2, 2, 1>();
-    ap_float_sincos<8, 18, 10, 20>();
-    ap_float_sinpi<3, 6, 6, 6>();
-    ap_float_cospi<18, 40, 18, 40>();
-    ap_float_sincospi<9, 20, 11, 20>();
-    ap_float_asin<2, 4, 2, 8>();
-    ap_float_asinpi<11, 23, 11, 23>();
-    ap_float_acos<4, 9, 3, 10>();
-    ap_float_acospi<2, 5, 3, 4>();
-    ap_float_atan<12, 31, 12, 31>();
-    ap_float_atapin<1, 38, 1, 32>();
-    ap_float_atan2<7, 16, 7, 17, 8, 18>();
-    ap_float_pow<8, 8, 9, 9, 10, 10>();
-    ap_float_powr<18, 35, 19, 35, 20, 35>();
-    ap_float_pown<4, 7, 10, 5, 9>();
-  });
-  return 0;
-}
+SYCL_EXTERNAL void fpga_ihs_float() {
+  ap_float_cast<11, 28, 9, 30>();
```

Comment:
Why not eliminate all the templates similarly to the changes in one of the files above?

Reply:
My bad, this was one of the earlier test cases I did, I didn't know I could do it as templates. It's fixed now!